javarosa-dag.rst

```diff
@@ -0,0 +1,304 @@
+Evaluation Logic Architecture
+##########################################
+
+JavaRosa uses a Direct Acyclic Graph (DAG) to manage triggerable XPath expressions and their dependencies.
+Those expressions can be put in the following attributes of the `bind <https://opendatakit.github.io/xforms-spec/#bindings>`_ section 
+in a form:
+
+* ``calculate``
+* ``relevant``
+* ``required``
+* ``readonly``
+
+The ``constraint`` attribute also takes an XPath expression but it is not managed by DAG.
+It is represented by ``Constraint`` class and evaluated before an answer is committed into the data model
+in the ``FormEntryController.answerQuestion(FormIndex index, IAnswerData data, boolean midSurvey)`` method.
+
+Classes
+-------
+
+- ``Triggerable`` --- An abstract class represents such expression, its context and targets which should be updated by this expression. It has two implementations:
```

Comment:
Perhaps there’s something like HTML’s <dl> that would format this section better than this approach with the three hyphens. Also, when I see multiple hyphens, I wonder if an em dash (—) would be better. Maybe @adammichaelwood can offer something.

Reply:
I agree that a definition list is preferable here. BUT...

These are Java classes, yes? Sphinx supports semantic authoring of software documentation (that's why we use it), so this stuff should ideally be documented using the Java domain. 

cf.:
 - Domains in Sphinx: http://www.sphinx-doc.org/en/stable/domains.html
 - Java domain in Sphinx: http://bronto-javasphinx.readthedocs.io/en/latest/

Java Domain in Sphinx is a plugin which we haven't added yet (see issue #46)
This has not yet been done because I didn't expect to be housing this kind of developer documentation. But... we could prioritize it.


------

Side note... I truly think three hyphens in source should automatically render to em-dash. I might implement that on this repo, because I hate:

 - trying to remember which key combo is em dash, especially since I have remapped some of my keys for emacs
 - trying to guess which type of dash I'm looking at in fixed-width fonts

Reply:
--- mapping to —   : great!
- – and — in fixed-width font: true!